src/phpDocumentor/Phpdoc3XmlToArrayConverter.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace phpDocumentor;
+
+final class Phpdoc3XmlToArrayConverter implements XmlConverter
+{
+    /**
+     * @var \SimpleXMLElement
+     */
+    private $xml;
+
+    /**
+     * @param \SimpleXMLElement $xml
+     */
+    public function __construct(\SimpleXMLElement $xml)
+    {
+        $this->xml = $xml;
+    }
+
+    /**
+     * Converts the phpDocumentor3 configuration xml to an array.
+     *
+     * @return array
+     */
+    public function convert()
+    {
+        $phpDocumentor = $this->xml;
+
+        $versions = [];
+        $template = [];
+
+        foreach ($phpDocumentor->children() as $key => $value) {
+            switch ((string) $key) {
+                case 'version':
+                    $versions[(string) $value->attributes()->number] = $this->buildVersions($value);
+                    break;
+                case 'template':
+                    $template = $this->buildTemplate($value);
+                    break;
+                default:
+                    break;
+            }
+        }
+        $outputDirectory = ((string) $phpDocumentor->parser->target) ?: 'file://build/docs';
+        $cacheDirectory  = ((string) $phpDocumentor->parser->cache) ?: '/tmp/phpdoc-doc-cache';
+
+        $phpdoc3Array = [
+            'phpdocumentor' => [
+                'paths'     => [
+                    'output' => $outputDirectory,
+                    'cache'  => $cacheDirectory,
+                ],
+                'versions'  => $versions,
+                'templates' => $template,
+            ]
+        ];
+
+        return $phpdoc3Array;
+    }
+
+    /**
+     * Builds the versions part of the array from the phpDocumentor3 configuration xml
+     *
+     * @param \SimpleXMLElement $version
+     *
+     * @return array
+     */
+    private function buildVersions(\SimpleXMLElement $version)
+    {
+        $extensions = [];
+        foreach ($version->api->extensions->children() as $extension) {
+            if ((string) $extension !== '') {
+                $extensions[] = (string) $extension;
+            }
+        }
+
+        $ignoreHidden   = filter_var($version->api->ignore->attributes()->hidden, FILTER_VALIDATE_BOOLEAN);
+        $ignoreSymlinks = filter_var($version->api->ignore->attributes()->symlinks, FILTER_VALIDATE_BOOLEAN);
+
+        return [
+            'folder' => (string) $version->folder,
+            'api'    => [
+                'format'               => ((string) $version->api->attributes()->format) ?: 'php',
+                'source'               => [
+                    'dsn'   => (string) $version->api->source->attributes()->dsn,
+                    'paths' => ((array) $version->api->source->path) ?: ['src'],
+                ],
+                'ignore'               => [
+                    'hidden'   => ($ignoreHidden) ?: true,
```

Comment:
`If $ignoreHidden is true then return true, otherwise return true`?

I think you have an error in this line and the next